### PR TITLE
Fix EmailStr for Cython

### PR DESCRIPTION
**Pull request by koxudaxi**
## Change Summary

This PR fixes EmailStr which is in Compiled Pydantic binary.

## Related issue number

https://github.com/samuelcolvin/pydantic/issues/1070

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: changes/1126-koxudaxi.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix EmailStr for Cython with when used in Linux.
```
Comment (samuelcolvin) [inline changes/1126-koxudaxi.md]:
```suggestion
Fix `EmailStr` for Cython when used in Linux to accept instances of `EmailStr`
```

File: pydantic/networks.py [hunk 2]
```diff
@@ -298,11 +298,13 @@ def __get_validators__(cls) -> 'CallableGenerator':
...
-        yield str_validator
         yield cls.validate
 
     @classmethod
-    def validate(cls, value: str) -> 'NameEmail':
+    def validate(cls, value: Any) -> 'NameEmail':
+        if type(value) == cls:
+            return value
+        value = str_validator(value)
         return cls(*validate_email(value))
```
Comment (koxudaxi) [inline pydantic/networks.py]: @samuelcolvin 
> Looks good, could you please also fix NameEmail and add a test for that too.
I found a bug that the validator can't parse NameEmail. It means `NameEmail` filed doesn't accept an instance of `NameEmail`

File: pydantic/networks.py [hunk 3]
```diff
@@ -380,7 +382,7 @@ def validate(cls, value: NetworkType) -> Union[IPv4Network, IPv6Network]:
 pretty_email_regex = re.compile(r'([\w ]*?) *<(.*)> *')
 
 
-def validate_email(value: str) -> Tuple[str, str]:
+def validate_email(value: Union[str]) -> Tuple[Union[str], Union[str]]:
```
Comment (samuelcolvin) [inline pydantic/networks.py]:
```suggestion
def validate_email(value: Union[str]) -> Tuple[str, str]:
```
I think should be fine?

File: pydantic/networks.py [hunk 4]
```diff
@@ -278,7 +278,7 @@ def __get_validators__(cls) -> 'CallableGenerator':
         yield cls.validate
 
     @classmethod
-    def validate(cls, value: str) -> str:
+    def validate(cls, value: Union[str]) -> Union[str]:
```
Comment (samuelcolvin) [inline pydantic/networks.py]:
```suggestion
    def validate(cls, value: Union[str]) -> str:
```

File: changes/1126-koxudaxi.md [hunk 5]
```diff
@@ -0,0 +1 @@
+Fix `EmailStr` for Cython when used in Linux to accept instances of `EmailStr`
```
Comment (samuelcolvin) [inline changes/1126-koxudaxi.md]:
```suggestion
Fix `EmailStr` and `NameEmail` to accept instances of themselves in cython.
```

File: pydantic/networks.py (hunk 3 above)
Comment (koxudaxi) [reply pydantic/networks.py]: @samuelcolvin 
I'm sorry, I misunderstood.

**Comment by samuelcolvin** [comment]
great thank you.

**Pull request merged**